nixos/modules/services/development/zammad.nix

```
@@ -0,0 +1,217 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.zammad;
+  serviceConfig = {
+    Type = "simple";
+    Restart = "always";
+
+    User = "zammad";
+    Group = "zammad";
+    PrivateTmp = true;
+    StateDirectory = builtins.baseNameOf cfg.dataDir;
+    WorkingDirectory = cfg.dataDir;
+
+    EnvironmentFile = cfg.secretsFile;
+  };
+  env = {
+    RAILS_ENV = "production";
+    NODE_ENV = "production";
+    RAILS_SERVE_STATIC_FILES="true";
+    RAILS_LOG_TO_STDOUT="true";
+  };
+in {
+
+  options = {
+    services.zammad = {
+      enable = lib.mkEnableOption "Zammad, a web-based, open source user support/ticketing solution.";
+
+      package = lib.mkOption {
+        type = lib.types.package;
+        default = pkgs.zammad;
+        defaultText = "pkgs.zammad";
+        description = "Zammad package to use.";
+      };
+
+      dataDir = lib.mkOption {
+        type = lib.types.path;
+        default = "/var/lib/zammad";
+        description = ''
+          Path to a folder that will contain Zammad working directory.
+        '';
+      };
+
+      host = lib.mkOption {
+        type = lib.types.str;
+        default = "127.0.0.1";
+        example = "192.168.23.42";
+        description = "Host address.";
+      };
+
+      port = lib.mkOption {
+        type = lib.types.int;
```

```suggestion
        type = lib.types.port;
```
(also for other port options)